patches/rocm-6.1.2/pytorch_vision/0003-pytorch-vision-build-fix-for-python311.patch

```diff
@@ -0,0 +1,31 @@
+From 26fc74a026e96b06a7bfbe3d4d435551c08b5a0e Mon Sep 17 00:00:00 2001
+From: Daniele <57776841+daniandtheweb@users.noreply.github.com>
+Date: Wed, 19 Jun 2024 15:42:08 +0200
+Subject: [PATCH 3/3] pytorch-vision-build-fix-for-python311
+
+Fix provided by @jeroen-mostert on https://github.com/lamikr/rocm_sdk_builder/issues/66#issuecomment-2169018614
+
+Signed-off-by: Daniele <57776841+daniandtheweb@users.noreply.github.com>
+Signed-off-by: Jeroen Mostert <jeroen.mostert@cm.com>
+---
+ setup.py | 4 +---
+ 1 file changed, 1 insertion(+), 3 deletions(-)
+
+diff --git a/setup.py b/setup.py
+index ce1cd90ca0..1df620fe83 100644
+--- a/setup.py
++++ b/setup.py
+@@ -383,9 +383,7 @@ def get_extensions():
+         ffmpeg_include_dir = os.path.join(ffmpeg_root, "include")
+         ffmpeg_library_dir = os.path.join(ffmpeg_root, "lib")
+ 
+-        gcc = os.environ.get("CC", shutil.which("gcc"))
```

Comment:
This is a little crude; my suggestion was actually (apologies for not making this a pull) using the `is_rocm_pytorch` flag set earlier:

```python
platform_tag = ""
if not is_rocm_pytorch:
    platform_tag = subprocess.run(...
    ...

if platform_tag:
    ...
```

Of course stubbing out the whole `platform_tag` logic also works if we know we never want to build this with gcc, but then all of the logic should simply be deleted, including the `if` block (`platform_tag` is not used outside this block, this appears to be logic specifically for Debian only). This should definitely be tested on Debian-derived systems to ensure it doesn't actually break anything that Debian needs. If the build outputs `Building torchvision without ffmpeg support` even when ffmpeg is installed that would be a bad sign.

Using the compiler to get the architecture seems suspect to begin with and there's probably a more elegant way that doesn't involve invoking gcc.

Reply:
Ohh, I didn't understand how you were thinking to implement it so I just went straight to the fastest fix.
I'll test it later, thanks for the explanation.